src/contact.rs

```diff
@@ -1451,4 +1476,27 @@ mod tests {
         assert!(addr_cmp(" aa@aa.ORG ", "AA@AA.ORG"));
         assert!(addr_cmp(" mailto:AA@AA.ORG", "Aa@Aa.orG"));
     }
+
+    #[test]
+    fn test_name_in_address() {
+        let t = dummy_context();
+
+        let contact_id = Contact::create(&t.ctx, "", "<dave@example.org>").unwrap();
+        let contact = Contact::load_from_db(&t.ctx, contact_id).unwrap();
+        assert_eq!(contact.get_name(), "");
+        assert_eq!(contact.get_addr(), "dave@example.org");
+
+        let contact_id = Contact::create(&t.ctx, "", "Mueller, Dave <dave@example.org>").unwrap();
+        let contact = Contact::load_from_db(&t.ctx, contact_id).unwrap();
+        assert_eq!(contact.get_name(), "Dave Mueller");
+        assert_eq!(contact.get_addr(), "dave@example.org");
```

Comment:
maybe also add a test that checks that the name given as parameter takes precedence over the name from the email address, so `create("name1", "name2 <addr@example.org>")` results in `name2`.

Reply:
Actually, currently, currently `create("name1", "name2 <addr@example.org>")` should result in `name1` (I think) (which might be what you meant because name1 is the name given as parameter and takes precedence).

Reply:
sure, i did a typo, it should result in `name1`.